tests/main/core16-provided-by-core/task.yaml

```diff
@@ -0,0 +1,26 @@
+summary: Ensure that the core snap is usable as an alias for core16
+
+execute: |
+    echo "Ensure core is installed"
+    if ! snap list core; then
+        snap install core
+    fi
+
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB/snaps.sh"
+    install_local test-snapd-sh-core16
+
+    echo "and core16 was not pulled in"
+    ! snap list core16
+
+    echo "and the snap works fine"
+    test-snapd-sh-core16.sh -c "echo hello" | MATCH hello
+    SNAP_CONFINE_DEBUG=1 test-snapd-sh-core16.sh -c "true" 2>&1 | MATCH "falling back to core instead of unavailable core16 snap"
+
+
```

Comment:
I think this warrants one comment about why snap-discard-ns is not required here:
```suggestion
# The missing base snap fallback check is performed on each startup, even if the mount namespace is reused.
# Therefore we don't have to discard the mount namespace constructed by the calls above.
```